school_hostel/models/hostel.py

```diff
@@ -0,0 +1,545 @@
+        room""",
+    )
+    guest_sofa = fields.Boolean(
+        "Guest sofa-bed",
+        help="""Activate/deactivate as per the
+        availability of guest sofa-bed in the room""",
+    )
+    tv = fields.Boolean(
+        "Television",
+        help="""Activate/deactivate as per the
+        availability of TV in the room""",
+    )
+    internet = fields.Boolean(
+        "Internet Access",
+        help="""Activate/deactivate
+       as per the availability of Internet-connectivity
+       in the room""",
+    )
+    refrigerator = fields.Boolean(
+        "Refrigerator",
+        help="""Activate/deactivate
+        as per the availability of Refrigerator
+        in the room""",
+    )
+    microwave = fields.Boolean(
+        "Microwave",
+        help="""Activate/deactivate as per
+        the availability of Microwave in the room""",
+    )
+    student_ids = fields.One2many(
+        "hostel.student", "room_id", string="Students", help="Enter students"
+    )
+
+    _sql_constraints = [
+        ("room_no_unique", "unique(room_no)", "Room number must be unique!"),
+        (
+            "floor_per_hostel",
+            "check(floor_no < 10)",
+            "Error ! Floor per HOSTEL should be less than 10.",
+        ),
+        (
+            "student_per_room_greater",
+            "check(student_per_room < 10)",
+            "Error ! Student per room should be less than 10.",
+        ),
+    ]
+
+    @api.model
+    def fields_view_get(
+        self, view_id=None, view_type="form", toolbar=False, submenu=False
+    ):
+        """Inherited this method to make edit button invisible in list view"""
+        res = super(HostelRoom, self).fields_view_get(
+            view_id=view_id,
+            view_type=view_type,
+            toolbar=toolbar,
+            submenu=submenu,
+        )
+        user_group = self.env.user.has_group("school_hostel.group_hostel_user")
+        doc = etree.XML(res["arch"])
+        if user_group:
+            if view_type == "tree":
+                nodes = doc.xpath("//tree[@name='hostel_room']")
+                for node in nodes:
+                    node.set("edit", "false")
+                res["arch"] = etree.tostring(doc)
+            if view_type == "form":
+                nodes = doc.xpath("//form[@name='hostel_room']")
+                for node in nodes:
+                    node.set("edit", "false")
+                res["arch"] = etree.tostring(doc)
+        return res
+
+    @api.constrains("rent_amount")
+    def _check_rent_amount(self):
+        """Constraint on negative rent amount"""
+        if self.rent_amount < 0:
+            raise ValidationError(
+                _(
+                    """
+                    Rent Amount Per Month should not be a negative value!"""
+                )
+            )
+
+
+class HostelStudent(models.Model):
+    _name = "hostel.student"
+    _description = "Hostel Student Information"
+    _rec_name = "student_id"
+
```

Comment:
why we need a new model here?
Typically it is a transaction for student.
Think optimal for the design.